src/Features/Core/Portable/LanguageServices/SymbolDisplayService/AbstractSymbolDisplayService.AbstractSymbolDescriptionBuilder.cs

```diff
@@ -175,6 +176,44 @@ private void AddExceptions(ISymbol symbol)
                 }
             }
 
+            protected abstract void AddCaptures(ISymbol symbol);
+
+            protected void AddCaptures(SyntaxNode syntax)
+            {
+                var semanticModel = GetSemanticModel(syntax.SyntaxTree);
+                if (semanticModel.IsSpeculativeSemanticModel)
+                {
+                    // Speculative semantic models cannot do region analysis
```

Comment:
@CyrusNajmabadi @dotnet/roslyn-ide  I'm not very familiar with speculation, so I'm not sure the impact of doing this (listing no captures when in speculative mode).

How often do we do QuickInfo in speculative mode? 
Should I list "Variables Captured: <unsaved document>" instead of removing the section?

Reply:
Sorry, can you catch me up on things.  Is there a reason the speculative semantic model doesn't have capture information?

--

> How often do we do QuickInfo in speculative mode?

basically, we *always* try.  The idea of using speculative models is to get good perf for features that don't need things like exact positions of elements in source code.  Specifically, as a user is editing in a method, we'll commonly use a previously cached semantic model and then speculate against it using the current method body.  This means far less binding and other compiler overhead for many features that would not be able to tell the difference.

--

IMO, if we can't get speculative models to support providing capture info, then i think it might be acceptable to consider making quickinfo no longer use speculative models.  Quick info has far different perf concerns versus intellisense (teh major speculative benefactor).  People are typing fast, and intellisense needs to be as fast as it can to not slow that down.  Quick info is very different.  It's something a user needs to go and manually do with the mouse after typing.  That time spent interacting with the mouse means it's far more likely the normal semantic model will be ready, and the benefits of speculation will be minimal.

Reply:
@jcouv Is there a deep reason that this API can't support speculation? Or is it mostly a paranoia check since we didn't want to test it?

Reply:
The speculative semantic model isn't modeling inserting the code at any particular position in the enclosing code, but merely somewhere within a given scope. You can point anywhere within a scope to speculate within that scope. We don't attempt to make sense of the speculative code as if it were somehow inserted into the flow of the program "at that point", as that's not how the API is supposed to work. If you want to imagine the enclosing code with some modification, you need to take that code and apply the modification you want. It would only make sense to do flow analysis in speculative code when you are speculating on a complete replacement for the method body.

Reply:
Here's what I gathered from discussion with @gafter:

Flow analysis does not make sense when speculating.
Speculating is asking semantic model questions about a speculative snippet in the context of a given *scope* of existing code. Although the scope is specified by pointing into the existing code, speculation only reasons about the scope that position is in.
Within that scope, we don't know (and shouldn't assume) where the speculative snippet might be inserted, so flow analysis could not be done as the code (and therefore flow and regions) is undetermined.

Reply:
From further testing in the IDE, it turns out I can hit the speculative case (shown below). Once I get that message, I'm not even sure how to get back to non-speculative case (this message seems very sticky 😞).
I think this feature will depend on removing speculation from QuickInfo as Cyrus suggested. I'll look over the weekend, but I may need some help with that.

![image](https://user-images.githubusercontent.com/12466233/35172564-04ff2f78-fd1d-11e7-93a1-0759b1297da0.png)